>From e44f7f4b889f8e0113ab1441fbb9c5bbf8a389bc Mon Sep 17 00:00:00 2001
From: Dominik Schulz <dominik.schulz@gauner.org>
Date: Mon, 3 Jun 2013 16:49:15 +0200
Subject: [PATCH] Add Acquire::http::SendRange option

This commit adds the option Acquire::http::SendRange which allows for
disabling sending of Range/If-Range headers. Support for these headers
is not fully implemented and may break the update process. By providing
this option the operator gains control over this behavior while
not changing any defaults for the majority of our users.
---
 methods/http.cc |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/methods/http.cc b/methods/http.cc
index db1085a..baaf7e9 100644
--- a/methods/http.cc
+++ b/methods/http.cc
@@ -740,7 +740,10 @@ void HttpMethod::SendReq(FetchItem *Itm,CircleBuf &Out)
 
    // Check for a partial file
    struct stat SBuf;
-   if (stat(Itm->DestFile.c_str(),&SBuf) >= 0 && SBuf.st_size > 0)
+   if (
+      _config->FindB("Acquire::http::SendRange", true) == true
+      && stat(Itm->DestFile.c_str(),&SBuf) >= 0 && SBuf.st_size > 0
+   )
    {
       // In this case we send an if-range query with a range header
       sprintf(Buf,"Range: bytes=%lli-\r\nIf-Range: %s\r\n",(long long)SBuf.st_size - 1,
-- 
1.7.9.5

